tests/system/test_list_rows.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import decimal
+
+from google.cloud import bigquery
+from google.cloud.bigquery import enums
+
+
+def test_list_rows_empty_table(bigquery_client: bigquery.Client, table_id: str):
+    from google.cloud.bigquery.table import RowIterator
+
+    table = bigquery_client.create_table(table_id)
+
+    # It's a bit silly to list rows for an empty table, but this does
+    # happen as the result of a DDL query from an IPython magic command.
+    rows = bigquery_client.list_rows(table)
+    assert isinstance(rows, RowIterator)
+    assert tuple(rows) == ()
+
+
+def test_list_rows_page_size(bigquery_client: bigquery.Client, table_id: str):
+    num_items = 7
+    page_size = 3
+    num_pages, num_last_page = divmod(num_items, page_size)
+
+    to_insert = [{"string_col": "item%d" % i, "rowindex": i} for i in range(num_items)]
+    bigquery_client.load_table_from_json(to_insert, table_id).result()
+
+    df = bigquery_client.list_rows(
+        table_id,
+        selected_fields=[bigquery.SchemaField("string_col", enums.SqlTypeNames.STRING)],
+        page_size=page_size,
+    )
+    pages = df.pages
+
+    for i in range(num_pages):
+        page = next(pages)
+        assert page.num_items == page_size
+    page = next(pages)
+    assert page.num_items == num_last_page
+
+
+def test_list_rows_scalars(bigquery_client: bigquery.Client, scalars_table: str):
+    rows = list(
+        sorted(
+            bigquery_client.list_rows(scalars_table), key=lambda row: row["rowindex"]
+        )
+    )
```

Comment:
`sorted()` already returns a list, no need for explicitly "listifying" it.

Reply:
D'oh! I wonder why I learned it as an iterable.